architectures/decentralized/solana-client/src/main.rs

```diff
@@ -341,9 +359,27 @@ async fn async_main() -> Result<()> {
                 CommitmentConfig::confirmed(),
             )
             .unwrap();
+
+            if treasurer_index.is_some() && treasurer_collateral_mint.is_none() {
```

Comment:
How does one determine their `treasurer_index`?

Reply:
We either hand-pick it by specifying it through the flag or we let the CLI auto-generate one for us from the run_id's hash. TLDR: we can pick a value and stick to it